### PR TITLE
Fix menus for macOS

### PR DESCRIPTION
**Pull request by Gadiguibou**
Turns out some of the menus (like the help menu) need to be handled differently for macOS because they're system menus that are expected to be present and standard for most apps.

I used [this example]( https://www.electronjs.org/docs/latest/api/menu#examples)from the electron docs as a template.

On macOS before:

https://github.com/Project-Fungus/fungus-gui/assets/34945306/d90d9edc-9874-479a-b161-fc04a080dcb9

On macOS after:

https://github.com/Project-Fungus/fungus-gui/assets/34945306/70f0e7bc-8f1e-44ec-8cd9-a896d6a5df4b

### CODE REVIEW

File: src/main.js [hunk 1]
```diff
@@ -30,72 +30,150 @@ app.whenReady().then(() => {
...
+                ...(isMac
+                    ? [
+                        { role: "pasteAndMatchStyle" },
+                        { role: "delete" },
+                        { role: "selectAll" },
+                        { type: "separator" },
+                        {
+                            label: "Speech",
+                            submenu: [
+                                { role: "startSpeaking" },
```
Comment (louis-hildebrand) [inline src/main.js:106]: Is this for navigating the UI with speech? Does it actually work?

File: src/main.js [hunk 2]
```diff
@@ -30,72 +30,150 @@ app.whenReady().then(() => {
...
-                { role: "toggleDevTools" },
-            ]
+                { role: "togglefullscreen" },
+            ],
+        },
+        // { role: 'windowMenu' }
+        {
+            label: "Window",
+            submenu: [
+                { role: "minimize" },
```
Comment (louis-hildebrand) [inline src/main.js:154]: The default keyboard shortcuts for minimize and close (CTRL+M and CTRL+W) clash with the ones I chose for opening the matches and warning views. Maybe we should change the latter to CTRL+SHIFT+M and CTRL+SHIFT+W or something.

File: src/main.js [hunk 3]
```diff
@@ -30,72 +30,150 @@ app.whenReady().then(() => {
...
         {
-            label: "&Help",
-            click: (_, browserWindow) => {
-                browserWindow.webContents.send("show-help-view");
-            }
-        }
+            role: "help",
+            submenu: [
+                {
+                    label: "Get Help?",
```
Comment (louis-hildebrand) [inline src/main.js:170]: Is it possible to add a keyboard shortcut here? Before I had ALT+H, which is what it says on the help page.

**Review by louis-hildebrand — CHANGES_REQUESTED** [review]
The help menu works for me as well, so that's good! There are a few small issues that should be easy to fix.

One thing I noticed in the videos is that the app name in the menu is just "Electron." According to this StackOverflow page, to fix that you need to set the `productName` in package.json and package the app (you can do that with `npm run make`). That would be nice, as long as it doesn't take more than a few minutes.

And finally, since the menu changed, parts of the help page are out of date. I don't mind updating it once all the other changes are made.

**Comment by Gadiguibou** [comment]
> One thing I noticed in the videos is that the app name in the menu is just "Electron." According to [this StackOverflow page](https://stackoverflow.com/questions/41551110/unable-to-override-app-name-on-mac-os-electron-menu), to fix that you need to set the productName in package.json and package the app (you can do that with npm run make). That would be nice, as long as it doesn't take more than a few minutes.

That was just for testing it works with the packaged app as well.

File: src/main.js (hunk 1 above)
Comment (Gadiguibou) [reply src/main.js:106]: Yes it does :)

File: src/main.js (hunk 2 above)
Comment (Gadiguibou) [reply src/main.js:154]: Sure! Fixed it

File: src/main.js (hunk 3 above)
Comment (Gadiguibou) [reply src/main.js:170]: Sure I'll go with alt-h for windows and Cmd+? for mac since these seem to be the standard.

Comment (Gadiguibou) [reply src/main.js:170]: Fixed in latest commit.

**Review by louis-hildebrand — APPROVED** [review]
Looks good to me. Thanks for taking care of this!

**Pull request merged**